### ISSUE TITLE
save_hyperparameters does not save result of LR finder

### ISSUE THREAD

**Issue by tbenst** [PyTorchLightning/pytorch-lightning#3522]
## 🐛 Bug
Follow-on to https://github.com/PyTorchLightning/pytorch-lightning/issues/3494. Because `save_hyperparameters` only saves the init args, it does not save the output of the learning rate finder, which mutates `lr`. These are two different bugs that result from the same implementation decision of basing hyperparams on init args. I think this is reasonable for sure, but perhaps the current value of `self.myarg` should be logged instead? or at least support passing a list of strings that are then looked up eg `['lr', 'num_filters']` -> `[self.lr, self.num_filters]`?

**Comment by Borda** [comment]
@tbenst mind send a draft PR with your suggestion so we are clear what you propose to change?
